ics23.go/test_helpers.go

```diff
@@ -0,0 +1,108 @@
+package iavl
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"sort"
+
+	"github.com/tendermint/iavl"
+	db "github.com/tendermint/tm-db"
+)
+
+// Result is the result of one match
+type Result struct {
+	Key      []byte
+	Value    []byte
+	Proof    *iavl.RangeProof
+	RootHash []byte
+}
+
+// GenerateResult makes a tree of size and returns a range proof for one random element
+//
+// returns a range proof and the root hash of the tree
+func GenerateResult(size int, loc Where) (*Result, error) {
+	tree, allkeys, err := BuildTree(size)
+	if err != nil {
+		return nil, err
+	}
+	key := GetKey(allkeys, loc)
+
+	value, proof, err := tree.GetWithProof(key)
+	if err != nil {
+		return nil, err
+	}
+	if value == nil {
+		return nil, fmt.Errorf("tree.GetWithProof returned nil value")
+	}
+	if len(proof.Leaves) != 1 {
+		return nil, fmt.Errorf("tree.GetWithProof returned %d leaves", len(proof.Leaves))
+	}
+	root := tree.Hash()
+
+	res := &Result{
+		Key:      key,
+		Value:    value,
+		Proof:    proof,
+		RootHash: root,
+	}
+	return res, nil
+}
+
+// Where selects a location for a key - Left, Right, or Middle
+type Where int
+
+const (
+	Left Where = iota
+	Right
+	Middle
+)
+
+// GetKey this returns a key, on Left/Right/Middle
+func GetKey(allkeys [][]byte, loc Where) []byte {
+	if loc == Left {
+		return allkeys[0]
+	}
+	if loc == Right {
+		return allkeys[len(allkeys)-1]
+	}
+	// select a random index between 1 and allkeys-2
+	idx := rand.Int()%(len(allkeys)-2) + 1
```

Comment:
```
G404: Use of weak random number generator (math/rand instead of crypto/rand) (gosec)
```

Reply:
Yeah, this is only for test code. It should probably be turned into a `_test.go` file.